src/Gravatar.php

```diff
@@ -86,8 +152,85 @@ private function createEmailHash(string $email): string
         return md5(strtolower(trim($email)));
     }
 
+    private function buildOptions(array $options): array
+    {
+        // Image size
+        if (array_key_exists('s', $options)) {
+            $size = filter_var($options['s'], FILTER_VALIDATE_INT);
+
+            if ($size === false || $size < self::MINIMUM_IMAGE_SIZE || $size > self::MAXIMUM_IMAGE_SIZE) {
+                unset($options['s']);
```

Comment:
I can see from the tests also that this silently hides any errors. I am not sure that I like it... maybe we should introduce an Exception inside the package. @sagikazarmark  what do you think?

Reply:
Yeah, silently failing probably doesn't add any value, since (I think) Gravatar will do the same for you. I would probably think about making validation optional.

Reply:
Greetings; thank you for the feedback. I have updated the PR to have a fourth parameter on `avatar()` to turn validation on (off by default). I have updated the tests accordingly.

Reply:
Hello.. the goal was if validation is true then validate the options and throw an exception instead of unset. For now use the `\InvalidArgumentException`. If the validation is false skip the validation function as you already do.

Reply:
Code and tests have been updated to throw an exception.